### DISCUSSION TITLE
The default Firebase app does not exist. Make sure you call initializeApp()

### DISCUSSION THREAD

**Discussion by Geoffrey-Pliez** [firebase/firebase-tools#5910] [Q&A]
In production, App is no longer initialized.
Like this the script no longer passed in the firebase-frameworks/firebase-aware.
Are you having the same issue ?

**Comment by christhompsongoogle** [comment]
Hi [Geoffrey-Pliez](https://github.com/Geoffrey-Pliez), I haven't seen that problem - can you share more information on what platform and SDK you're using?

**Comment by Geoffrey-Pliez** [comment]
Hi @christhompsongoogle,
I'm working on Windows and using the latest js SDK.
It's an Astro project that uses web framework.
But when I deploy it, I get the following error:
[ssr] Error: The default Firebase app does not exist. Make sure you call initializeApp() before using any of the Firebase services.
As if the script no longer passed through firebase-aware.
I didn't have this problem before firebase-tools 12.x.x.

**Comment by christhompsongoogle** [comment]
It looks like the error is being thrown here:

https://cs.opensource.google/firebase-sdk/firebase-admin-node/+/master:src/app/lifecycle.ts;drc=9ab37f5d6ed6e03307869a3d46ef855b65666047;l=68

The location is in the admin SDK but I wonder if this is a web frameworks issue.

@lahirumaramba May be able to add more insight.

**Reply by lahirumaramba** [reply]
Sounds like this is related to web frameworks. Adding @DellaBitta and @jamesdaniels to triage. Thanks!

**Comment by DellaBitta** [comment]
Hi @Geoffrey-Pliez,

We'd like to better understand your app's setup and the flows that lead to this error. Could you create an Issue in this repo and provide the following information:

- You mentioned that this started to occur after using 12.x.x. Could you tell us which version worked, and which version of 12.x.x that caused the problem to occur? It would help us narrow down which changes may have caused the issue.
- You mentioned that this occurs after deployment. How are you deploying (are you using Firebase hosting)?
- Since the error occurs in the admin SDK, we were wondering if the error came from your use of the admin SDK or some internal usage. Are you invoking the admin SDK directly?
- Could you provide some source code to provide context as to where / when the error occurs, or a stand alone app that exhibits the behavior?

Thanks!

**Comment by Geoffrey-Pliez** [comment]
Hi @DellaBitta ,

[Here](https://stackblitz.com/edit/github-bi9q8b?file=src%2Fpages%2Findex.astro) is the minimal source code to reproduce the problem.

When I test locally with the prod_preview command (set FIREBASE_FRAMEWORKS_BUILD_TARGET=production&& npm run emulators), 
it works correctly and the code passes well in firebase-aware.

When I deploy on the server (firebase hosting) via the firebase deploy command, it no longer works and get the message 
` [ssr] Error: The default Firebase app does not exist. Make sure you call initializeApp() before using any of the Firebase services.`
and guess that on the server the firebase-aware script is no longer called.

I'm not sure but I think it's since upgrading from 11.30.0 to 12.0.0

**Comment by christhompsongoogle** [comment]
Thank you @Geoffrey-Pliez  for the reproducible example. Since this seems like it's going down the route of a bug-fix I'm going to open an issue and close this discussion.

https://github.com/firebase/firebase-tools/issues/5980

**Comment by cyrilmano** [comment]
Here is how I resolved the issue [HERE](https://github.com/firebase/firebase-tools/issues/5980#issuecomment-3035063328)
